.github/workflows/tests.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        run: npm i
```

Comment:
you should use `npm ci` rather than `npm i` as it will validate a predictable state.

Reply:
You should provide a lock btw.